### ISSUE TITLE
Error when fine-tuning Caffe model using DIGITS 

### ISSUE THREAD

**Issue by antran89** [NVIDIA/DIGITS#347]
Hi developers,
I am sorry, if I am posting in the wrong place. I searched user group, but did not find any clues. So, I report a bug here.
When I started fine-tuning a model using **bvlc_reference_caffenet** model, I followed tutorials of using Caffe to fine-tune **flickr_style** dataset. If I use Caffe, it was running fine, but it is hard to visualize the loss values. So, I decided to use DIGITS to fine tune my models on my data. The differenc is that I use the latest version of [DIGITS](https://github.com/NVIDIA/DIGITS) and [NVIDIA-Caffe](https://github.com/NVIDIA/caffe) to do fine-tuning. 
Then it generates this error

```
2015-10-07 14:14:02 [20151007-141401-e882] [INFO ] Train Caffe Model task started.
2015-10-07 14:14:08 [20151007-141401-e882] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated transformation parameters: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
2015-10-07 14:14:08 [20151007-141401-e882] [ERROR] Train Caffe Model: Note that future Caffe releases will only support transform_param messages for transformation fields.
2015-10-07 14:14:08 [20151007-141401-e882] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated V1LayerParameter: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
2015-10-07 14:14:09 [20151007-141401-e882] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated transformation parameters: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
2015-10-07 14:14:09 [20151007-141401-e882] [ERROR] Train Caffe Model: Note that future Caffe releases will only support transform_param messages for transformation fields.
2015-10-07 14:14:09 [20151007-141401-e882] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated V1LayerParameter: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
127.0.0.1 - - [2015-10-07 14:23:03] "POST /jobs/20151007-141401-e882/abort HTTP/1.1" 200 128 0.002188
```
Hope to receive your comments.

**Comment by howtobeahacker** [comment]
My train_val.prototxt and solver.prototxt are attached!

**Comment by lukeyeager** [comment]
Are you sure this is an error? Is this a duplicate of #140? See this comment - https://github.com/NVIDIA/DIGITS/issues/140#issuecomment-110545214.

**Comment by howtobeahacker** [comment]
Hi @lukeyeager 
I downloaded again the caffe model from the [script](https://github.com/BVLC/caffe/blob/master/scripts/download_model_binary.py) provided in the latest Caffe. The error is still the same. It did not progress after few minutes, so I aborted them. 
```
2015-10-08 23:37:03 [20151008-233702-fa10] [INFO ] Train Caffe Model task started.
2015-10-08 23:37:11 [20151008-233702-fa10] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated transformation parameters: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
2015-10-08 23:37:11 [20151008-233702-fa10] [ERROR] Train Caffe Model: Note that future Caffe releases will only support transform_param messages for transformation fields.
2015-10-08 23:37:11 [20151008-233702-fa10] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated V1LayerParameter: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
2015-10-08 23:37:15 [20151008-233702-fa10] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated transformation parameters: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
2015-10-08 23:37:15 [20151008-233702-fa10] [ERROR] Train Caffe Model: Note that future Caffe releases will only support transform_param messages for transformation fields.
2015-10-08 23:37:15 [20151008-233702-fa10] [ERROR] Train Caffe Model: Attempting to upgrade input file specified using deprecated V1LayerParameter: /home/tranlaman/BLVC-caffe/models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel
127.0.0.1 - - [2015-10-08 23:40:21] "POST /jobs/20151008-233702-fa10/abort HTTP/1.1" 200 128 0.004213
2015-10-08 23:40:26 [20151008-233702-fa10] [INFO ] Job deleted.
127.0.0.1 - - [2015-10-08 23:40:26] "DELETE /jobs/20151008-233702-fa10 HTTP/1.1" 200 128 0.004342
```

**Comment by lukeyeager** [comment]
Can you post the raw caffe log? I suspect it may just be really slow to respond.

**Comment by howtobeahacker** [comment]
Hi @lukeyeager.
This is my caffe log file [http://pastebin.com/DaC8WNLD]. I observe that `Data layer prefetch queue empty` is as the same as previous time.

**Comment by lukeyeager** [comment]
```
I1009 10:07:20.342645 14891 solver.cpp:314] Iteration 0, Testing net (#0)
I1009 10:07:20.354153 14899 blocking_queue.cpp:50] Waiting for data
I1009 10:07:20.405398 14897 blocking_queue.cpp:50] Waiting for data
I1009 10:07:20.405930 14899 blocking_queue.cpp:50] Waiting for data
I1009 10:07:20.467383 14897 blocking_queue.cpp:50] Waiting for data
I1009 10:07:20.486361 14897 blocking_queue.cpp:50] Waiting for data
I1009 10:07:20.511524 14891 blocking_queue.cpp:50] Data layer prefetch queue empty
...
I1009 10:16:44.137663 14891 blocking_queue.cpp:50] Data layer prefetch queue empty
I1009 10:16:44.168421 14899 blocking_queue.cpp:50] Waiting for data
I1009 10:16:44.213986 14899 blocking_queue.cpp:50] Waiting for data
I1009 10:16:44.366152 14899 blocking_queue.cpp:50] Waiting for data
I1009 10:16:44.383676 14899 blocking_queue.cpp:50] Waiting for data
I1009 10:16:44.427858 14899 blocking_queue.cpp:50] Waiting for data
```
How big is your dataset? 9 minutes seems like it should be plenty long enough to get some kind of a response, but if you're trying to train on all of ImageNet with a small GPU, I can see how the validation pass might take longer than that.

Can you try fine-tuning on a small dataset? Have you been able to use DIGITS to train any other model successfully on your machine?

**Comment by howtobeahacker** [comment]
@lukeyeager Hi,
For your information, I did training and following tutorial for NVIDIA Deep Learning course [https://developer.nvidia.com/deep-learning-courses]. It was successful with DIGITS in my local machine with NVIDIA-Caffe 0.12. I never did fine-tune with DIGITS before. Then I upgrade NVIDIA-Caffe to master branch and this error happened.
From what I read from [NVIDIA-Caffe#41](https://developer.nvidia.com/deep-learning-courses), it looks like the problem with post 0.12 version of NVIDIA-Caffe.

**Comment by howtobeahacker** [comment]
Hi @lukeyeager 
I can confirm that the same thing happened when fine-tuning even on the latest version of BVLC-Caffe. After a few iterations, Caffe will shutdown my workstation. I have repeated the errors multiple times and still don't know why. I might be because of my train_val.prototxt file or Caffe binary model :sweat:. I do not why I succeeded in the first time.
I will try to localize the problem and report this error to Caffe developers.

**Comment by howtobeahacker** [comment]
Hi @lukeyeager 
I took your advice and tried fine-tuning on Flickr example provided in Caffe. Something serious problem occurred in my computer.
Could you please execute for me the Flickr fine-tune examples in [NVIDIA-Caffe](https://github.com/NVIDIA/caffe/tree/master/examples/finetune_flickr_style) or [BLVC-Caffe](https://github.com/BVLC/caffe/tree/master/examples/finetune_flickr_style)?
Whenever I run it, after sometimes, it will restart my workstation whether I run it on CPU or GPU. Please help me to confirm whether it is the same on your machine.
Thanks,

**Comment by gsxuan1127** [comment]
I have meet the same problem as 
I1009 10:16:44.137663 14891 blocking_queue.cpp:50] Data layer prefetch queue empty
I1009 10:16:44.168421 14899 blocking_queue.cpp:50] Waiting for data
And after a short time waiting, the ubuntu system shutted down.
I had do a lot of trys, the problem is still not solved.

**Comment by deepsemantic** [comment]
Hi @antran89 and @gsxuan1127 ,
I have same error, but the different is my ubuntu did not shut down. Do you guys already know how to fix this problem? I am using the caffe I downloaded one week ago.

**Comment by antran89** [comment]
Hi @deepsemantic, @gsxuan1127, @lukeyeager,
Sorry, I did not close this issue previously. After sometime, I figured out that the restarting problem is related to power issues. It might be that your workstation does not support enough power for GPUs (i.e., in my cases, Titan X). 
Please check your power issues, power cables, and try disable [GPU auto-boost](http://devblogs.nvidia.com/parallelforall/increase-performance-gpu-boost-k80-autoboost/) if it does not solve your problem.
Here is my summarization of how to set up proper workstation for Deep Learning https://drive.google.com/open?id=1McEMHewtMOwhIyUSO_2AZxbeEwRihjF2Ql_eYQ6bq2o.

**Comment by lukeyeager** [comment]
Nice presentation @antran89 - thanks for the link!
